examples/tutorials/07_vpp-connection/main.go

```diff
@@ -0,0 +1,169 @@
+	a := agent.NewAgent(agent.AllPlugins(p))
+
+	// Run starts the agent with plugins, wait until shutdown
+	// and then stops the agent and its plugins.
+	if err := a.Run(); err != nil {
+		log.Fatalln(err)
+	}
+}
+
+// HelloWorld represents our plugin.
+type HelloWorld struct {
+	vppChan api.Channel
+
+	GoVPPMux govppmux.API
+}
+
+// String is used to identify the plugin by giving it name.
+func (p *HelloWorld) String() string {
+	return "HelloWorld"
+}
+
+// Init is executed on agent initialization.
+func (p *HelloWorld) Init() (err error) {
+	log.Println("Hello World!")
+
+	if p.vppChan, err = p.GoVPPMux.NewAPIChannel(); err != nil {
+		panic(err)
+	}
+
+	p.syncVppCall()
+	p.asyncVppCall()
+	p.multiRequestCall()
+
+	return nil
+}
+
+func (p *HelloWorld) syncVppCall() {
+	// prepare request
+	request := &interfaces.CreateLoopback{
+		MacAddress: macParser("00:00:00:00:00:01"),
+	}
+	// prepare reply
+	reply := &interfaces.CreateLoopbackReply{}
+	// send request and obtain reply
+	err := p.vppChan.SendRequest(request).ReceiveReply(reply)
+	if err != nil {
+		panic(err)
+	}
+	// check return value
+	if reply.Retval != 0 {
+		log.Panicf("Sync call loopback create returned %d", reply.Retval)
+	}
+
+	log.Printf("Sync call created loopback with index %d", reply.SwIfIndex)
+}
+
+func (p *HelloWorld) asyncVppCall() {
+	// prepare requests
+	request1 := &interfaces.CreateLoopback{
+		MacAddress: macParser("00:00:00:00:00:02"),
+	}
+	request2 := &interfaces.CreateLoopback{
+		MacAddress: macParser("00:00:00:00:00:03"),
+	}
+
+	// obtain contexts
+	reqCtx1 := p.vppChan.SendRequest(request1)
+	reqCtx2 := p.vppChan.SendRequest(request2)
+
+	// wait a bit
+	time.Sleep(2 * time.Second)
+
+	// prepare replies
+	reply1 := &interfaces.CreateLoopbackReply{}
+	reply2 := &interfaces.CreateLoopbackReply{}
+
+	// receive replies
+	if err := reqCtx1.ReceiveReply(reply1); err != nil {
+		panic(err)
+	}
+	if err := reqCtx2.ReceiveReply(reply2); err != nil {
+		panic(err)
+	}
+
+	// check return values
+	if reply1.Retval != 0 {
+		log.Panicf("Async call loopback1 create returned %d", reply1.Retval)
+	}
+	if reply2.Retval != 0 {
+		log.Panicf("Async call loopback2 create returned %d", reply2.Retval)
+	}
+
+	log.Printf("Async call created loopbacks with indexes %d and %d",
+		reply1.SwIfIndex, reply2.SwIfIndex)
+}
+
+func (p *HelloWorld) multiRequestCall() {
+	// prepare request
+	request := &interfaces.SwInterfaceDump{}
+	multiReqCtx := p.vppChan.SendMultiRequest(request)
+
+	// read replies in the loop
+	for {
+		reply := &interfaces.SwInterfaceDetails{}
+		last, err := multiReqCtx.ReceiveReply(reply)
+		if err != nil {
+			panic(err)
+		}
+		if last {
+			break
+		}
+		log.Printf("received VPP interface with index %d", reply.SwIfIndex)
+	}
+}
+
+// AfterInit is executed after initialization of all plugins. It's optional
+// and used for executing operations that require plugins to be initalized.
+func (p *HelloWorld) AfterInit() error {
```

Comment:
Try to omit this if not used. Shorter examples = less code to read = easier to understand.

Reply:
will be used as you noted below

Reply:
yes, its based on the previous tutorial so I wanted to left original code untouched so it won't confuse anybody.. but I guess you're right, I'll remove it